src/app/course/add-page/course-durarion-input/course-duration-input.component.ts

```diff
@@ -1,12 +1,33 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, forwardRef, OnInit } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
 
 @Component({
   selector: 'app-course-duration-input',
   templateUrl: './course-duration-input.component.html',
-  styleUrls: ['./course-duration-input.component.scss']
+  styleUrls: ['./course-duration-input.component.scss'],
+  providers: [{
+    provide: NG_VALUE_ACCESSOR,
+    multi: true,
+    useExisting: forwardRef(() => CourseDurationInputComponent),
+  }]
 })
-export class CourseDurationInputComponent  {
-
+export class CourseDurationInputComponent implements OnInit, ControlValueAccessor {
   public duration;
+  onChange: (value) => void;
+  onTouched: () => void;
+
+  constructor() { }
+
+  ngOnInit() {}
+
+  writeValue(value): void {
+    this.duration = value;
+  }
+  registerOnChange(fn: any): void {
+    this.onChange = fn;
```

Comment:
You don't call `onChange` callback when data is changed by user inside this component. That's why new duration is not passed to the parent form